WorkflowDemos/WorkflowDemosTest/src_test/test/bpm/TestProcurementRequest.java

```diff
@@ -0,0 +1,225 @@
+package test.bpm;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.util.List;
+
+import org.awaitility.Awaitility;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import ch.ivyteam.ivy.bpm.engine.client.BpmClient;
+import ch.ivyteam.ivy.bpm.engine.client.ExecutionResult;
+import ch.ivyteam.ivy.bpm.engine.client.UiMocker;
+import ch.ivyteam.ivy.bpm.exec.client.IvyProcessTest;
+import ch.ivyteam.ivy.process.model.value.PID;
+import ch.ivyteam.ivy.security.exec.Sudo;
+import ch.ivyteam.ivy.workflow.CaseState;
+import ch.ivyteam.ivy.workflow.ICase;
+import ch.ivyteam.ivy.workflow.ITask;
+import ch.ivyteam.ivy.workflow.TaskState;
+import workflow.humantask.ProcurementRequest;
+
+/**
+ * Early preview of {@link BpmClient} testing API usage. 
+ * 
+ * Here to expose difficulties when working with workflow centered processes.
+ * In order to improve until the release 9.1
+ *
+ * Tests the ProcurementRequest:
+ * <code>WorkflowDemos/processes/Humantask/ProcurementRequestParallel.mod</code>
+ * 
+ * @author rew
+ * @since Mar 3, 2020
+ */
+@SuppressWarnings("all")
+@IvyProcessTest
+class TestProcurementRequest
+{
+  private final ProcurementRequest testData = new ProcurementRequest();
+  
+  @BeforeEach
+  void before()
+  {
+    testData.setDescription("PC");
+    testData.setAmount(3);
+    testData.setPricePerUnit(1222.95);
+  }
+  
+  /**
+   * Manager and Team Leader do not verify
+   */
+  @Test
+  void approvalTask_notVerified_byBoth(BpmClient bpmClient, UiMocker ui) throws Exception
+  {
+    ui.mock(new PID("15254DC87A1B183B-f3"), in -> testData);
```

Comment:
consider having a shared constant for these PIDs which are frequently used in this test:
```
static interface Element{
  PID verifyTAsk = new PID(XVCSDF);
  PID another ...
}
```
this would make the test highly readable even though we have yet no cool api to lookup process elements.